### DISCUSSION TITLE
Graphql api get commit messages user made to all repos and branches for this weak

### DISCUSSION THREAD

**Discussion by RudolfHladik** [community/community#24830] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I want to get to commit messages I made this week throughout every repo and branch.</p>
<p>I can find repos which I contributed to from some date with `contributionsCollection(from: $fromDate) {</p>
<p>commitContributionsByRepository</p>
<p>}</p>
<p>but I don’t know how to get name of the branch the commit was made to or the commit message or any reference to commit.</p>
<p>Thanks in advance.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/rudolfhladik">@rudolfhladik</a>,</p>
<p>Thank you for being here! You could try the following query for example:</p>
<pre><code>{
  search(query: "org:atom", type: REPOSITORY, last: 100) {
    nodes {
      ... on Repository {
        name
        defaultBranchRef {
          name
          target {
            ... on Commit {
              history(first: 100, since: "2019-10-11T00:00:00") {
                totalCount
                nodes {
                  ... on Commit {
                    committedDate
                    additions
                    messageHeadline
                    author {
                      name
                      email
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
</code></pre>
